src/ActionModal/utils/signer.js

```diff
@@ -11,57 +11,75 @@ export async function cancelSign(submitType = "", { address, network }) {
 
 export async function getSigner(
   config,
-  submitType = "",
-  { address, password, network }
+  signingType = "",
+  { address, password, network, networkType }
 ) {
-  if (submitType === `local`) {
-    const { signWithPrivateKey, getStoredWallet } = await import(
-      "@lunie/cosmos-keys"
-    )
-
+  if (signingType === `local`) {
+    const { getStoredWallet } = await import("@lunie/cosmos-keys")
     const wallet = getStoredWallet(address, password)
-    return signMessage => {
-      const signature = signWithPrivateKey(
-        signMessage,
-        Buffer.from(wallet.privateKey, "hex")
-      )
 
-      return {
-        signature,
-        publicKey: Buffer.from(wallet.publicKey, "hex")
-      }
+    switch (networkType) {
+      case "cosmos":
+        return await getCosmosLocalSigner(wallet)
+    }
+  } else if (signingType === `ledger`) {
+    switch (networkType) {
+      case "cosmos":
+        return await getCosmosLedgerSigner(config)
     }
-  } else if (submitType === `ledger`) {
-    // importing default here to be compatible with Jest
-    const { default: Ledger } = await import("@lunie/cosmos-ledger")
+  } else if (signingType === `extension`) {
+    return signMessage => {
+      return signWithExtension(signMessage, address, network)
+    }
+  }
 
-    return async signMessage => {
-      const ledger = new Ledger(config)
-      let publicKey, signature
-      try {
-        publicKey = await ledger.getPubKey()
-        signature = await ledger.sign(signMessage)
-      } catch (err) {
-        /* istanbul ignore next: specific error rewrite */
-        if (err.message.trim().startsWith("Device is already open")) {
-          throw new Error(
-            "Something went wrong connecting to your Ledger. Please refresh your page and try again."
-          )
-        }
-        throw err
-      }
+  throw new Error(
+    `Lunie doesn't support signing via ${signingType} for network type ${networkType}`
+  )
+}
 
-      // cleanup. if we leave this open, the next connection will brake for HID
-      ledger.cosmosApp.transport.close()
+async function getCosmosLocalSigner(wallet) {
+  const { signWithPrivateKey } = await import("@lunie/cosmos-keys")
 
-      return {
-        signature,
-        publicKey
+  return signMessage => {
+    const signature = signWithPrivateKey(
+      signMessage,
+      Buffer.from(wallet.privateKey, "hex")
+    )
+
+    return {
+      signature,
+      publicKey: Buffer.from(wallet.publicKey, "hex")
+    }
+  }
+}
+async function getCosmosLedgerSigner(config) {
+  // TODO show which properties of config are actually needed
```

Comment:
What properties of config? Why is this now needed?

Reply:
This was in the code before. This is just a refactor. `config` is passed through to the library. It would be nice to show here which properties of `config.js` are actually needed. This would help with search and replace or when we want to delete things from config.js.

Reply:
Oh right. Thank you so much for the clarification